src/MyEventEmitter.js

```diff
@@ -1,12 +1,200 @@
+/* eslint-disable no-console */
 'use strict';
 
-export class MyEventEmitter {
-  on() {}
-  once() {}
-  off() {}
-  emit() {}
-  prependListener() {}
-  prependOnceListener() {}
-  removeAllListeners() {}
-  listenerCount() {}
+class MyEventEmitter {
+  constructor() {
+    this.events = {};
+  };
+
+  validateInput(eventName = undefined, listener = undefined) {
+    return {
+      isNameValid: typeof eventName === 'string',
+      isListenerValid: typeof listener === 'function',
+    };
+  }
+
+  on(eventName, listener) {
+    const { events, validateInput } = this;
+    const { isNameValid, isListenerValid } = validateInput(eventName, listener);
+
+    if (!isNameValid && !isListenerValid) {
+      console.log(
```

Comment:
Use '||' instead of '&&' for checking invalid input: if (!isNameValid || !isListenerValid) { ... }